File: src/main/java/dealing/with/generalization/Customer.java
package dealing.with.generalization;

import lombok.AllArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@AllArgsConstructor
public class Customer {
    private final String name;
    private final String email;
    private final String phone;
    private final LocalDate creationDate;
    private final LocalDate conversionDate;
    private final List<String> messages;
}

File: src/scala/src/main/scala/dealing/with/generalization/Customer.scala
package dealing.`with`.generalization

import java.time.LocalDate

class Customer(
    name: String,
    email: String,
    phone: String,
    creationDate: LocalDate,
    val conversionDate: LocalDate,
    messages: List[String]
) extends Party(name, email, phone, creationDate, messages)
